test/integration/test_connection_management.py

```diff
@@ -0,0 +1,96 @@
+  args = [
+      fixture.getTestServerRootUri(), "--rps",
+      str(rps), "--duration", "1", "--request-header", "x-envoy-fault-delay-request:500",
+      "--max-pending-requests",
+      str(max_pending_requests), "--max-requests-per-connection",
+      str(requests_per_connection), "--connections",
+      str(number_of_connections)
+  ]
+  if h2:
+    args.append("--h2")
+  parsed_json, _ = fixture.runNighthawkClient(args)
+  counters = fixture.getNighthawkCounterMapFromJson(parsed_json)
+  if run_test_expectation:
+    assertCounterEqual(counters, "upstream_cx_total", expected_connections)
+  return counters
+
+
+# A series that tests with queueing disabled
+@pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
+def test_http_h1_connection_management_1(http_test_server_fixture):
+  run_with_number_of_connections(http_test_server_fixture, 1)
+
+
+@pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
+def test_http_h1_connection_management_2(http_test_server_fixture):
+  run_with_number_of_connections(http_test_server_fixture, 2)
+
+
+@pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
+def test_http_h1_connection_management_10(http_test_server_fixture):
+  run_with_number_of_connections(http_test_server_fixture, 10)
+
+
+# A series that tests with queueing enabled
+@pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
+def test_http_h1_connection_management_with_queue_1(http_test_server_fixture):
+  run_with_number_of_connections(http_test_server_fixture, 1, max_pending_requests=100)
+
+
+@pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
+def test_http_h1_connection_management_with_queue_2(http_test_server_fixture):
+  run_with_number_of_connections(http_test_server_fixture, 2, max_pending_requests=100)
+
+
+@pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
+def test_http_h1_connection_management_with_queue_10(http_test_server_fixture):
+  run_with_number_of_connections(http_test_server_fixture, 10, max_pending_requests=100)
+
+
+# Test h1 with a single request_per_connection
+@pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
+def test_http_h1_connection_management_single_request_per_conn_1(http_test_server_fixture):
+  counters = run_with_number_of_connections(
+      http_test_server_fixture,
+      1,
+      max_pending_requests=1,
+      requests_per_connection=1,
+      run_test_expectation=False)
+  requests = counters["upstream_rq_total"]
+  assertCounterBetweenInclusive(counters, "upstream_cx_total", requests, requests + 1)
+
+
+# Test h2 with a single request_per_connection
+@pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
+def test_http_h2_connection_management_single_request_per_conn_1(http_test_server_fixture):
+  counters = run_with_number_of_connections(
+      http_test_server_fixture,
+      1,
```

Comment:
Would it make sense to copy this to an additional test with a value other than 1, as a sanity check?

Reply:
yes, that makes sense to me, done in 0fcaf9185769ad68e07c51c9e3fee55e0035e6f3

Reply:
ugh, sorry, plus https://github.com/envoyproxy/nighthawk/pull/178/commits/f3ae31052b07d8108c385b6808bc85b720525daf (which makes us test h2 again)